pkg/controller/daemon/daemon_controller.go

```diff
@@ -1239,7 +1239,7 @@ func (dsc *DaemonSetsController) nodeShouldRunDaemonPod(node *v1.Node, ds *exten
 	wantToRun, shouldSchedule, shouldContinueRunning = true, true, true
 	// If the daemon set specifies a node name, check that it matches with node.Name.
 	if !(ds.Spec.Template.Spec.NodeName == "" || ds.Spec.Template.Spec.NodeName == node.Name) {
-		return false, false, false, nil
+		return false, false, false, fmt.Errorf("Spec.NodeName does not match node's name")
```

Comment:
This just means that the pod shouldn't run on that node. That's not an error. This breaks all usages of nodeShouldRunDaemonPod in this controller.